### PR TITLE
Handling prototype PTEs on Windows x64

### PR DESCRIPTION
**Pull request by thejanit0r**
Added handling of prototype PTEs on Windows x64.

The bug was found while dumping and analyzing user-mode memory regions of `ntdll.dll`.

Info about prototype PTEs:

- https://codemachine.com/articles/prototype_ptes.html
- Windows Internals 7th - P1

### CODE REVIEW

File: volatility3/framework/layers/intel.py [hunk 1]
```diff
@@ -464,6 +464,27 @@ def _translate_swap(
                     entry=excp.entry,
                     swap_offset=swap_offset,
                 )
+            # Handle Prototype PTEs
+            if isinstance(self, WindowsIntel32e):
+                if pbit:           
+                    vaddr = (0xffff << self._maxvirtaddr) | (entry >> 16)
```
Comment (ikelos) [inline volatility3/framework/layers/intel.py:470]: Also, are the `0xffff` in order to canonicalize the address?  There's a [separate function](https://github.com/volatilityfoundation/volatility3/blob/a4b927f478eca745a8a4c7ec25c77fa6de49d0df/volatility3/framework/layers/intel.py#L119) for doing that on intel layers...

File: volatility3/framework/layers/intel.py [hunk 2]
```diff
@@ -464,6 +464,27 @@ def _translate_swap(
...
+                    try:
+                        pte_addr = self._translate_swap(self, vaddr, bit_offset)
```
Comment (ikelos) [inline volatility3/framework/layers/intel.py:472]: Could this potentially enter an infinite loop given bad data?

**Review by ikelos — COMMENTED** [review]
Thanks, just a couple of points, but I'd also like to get this checked out by some others to make sure it functions as intended (since it's pretty core code).  I think @iMHLv2 is probably best placed to look at this one...

**Comment by thejanit0r** [comment]
I'll close the PR as I've come across some further issues (e.g. VAD hardware PTE), for which I rewrote some of the code for this PR. Thanks @ikelos for the quick review and feedback. I will follow up with a new PR.

**Pull request closed without merging**